### PR TITLE
Decrease inbound queue test gas & update snapshot

### PR DESCRIPTION
**Pull request by doubledup**
Using the `headProof.pos` field in the mock parachain client to avoid comparing whole proof structs and copying them from storage to calldata.

Gas consumption is generally lower since the last time we committed a snapshot, with these exceptions:
- InboundChannelTest:testSubmitShouldNotFailOnHandlerOOG (211741 -> 643392)
- NativeTokensTest:testCreateSuccessful (38508 -> 75682)
- NativeTokensTest:testHandleUnlockMessageSuccessful (125590 -> 151017)
- NativeTokensTest:testLockRevertsZeroAmount (11261 -> 11366)
- NativeTokensTest:testLockSuccessful (130192 -> 156773)
- TokenVaultTest:testInsufficientBalance (17117 -> 17294)
- TokenVaultTest:testNonOwnerCannotWithdraw (20672 -> 48935)
- TokenVaultTest:testWithdrawSuccessful (124729 -> 124731)

### CODE REVIEW

**Review by vgeddes — APPROVED** [review]
+1

File: core/packages/contracts/test/InboundQueue.t.sol [hunk 1]
```diff
@@ -21,14 +21,17 @@ contract InboundQueueTest is Test {
...
-    IParachainClient.Proof public parachainHeaderProof = IParachainClient.Proof(
-        new bytes(0),
-        new bytes(0),
-        IParachainClient.HeadProof(0, 0, new bytes32[](0)),
-        IParachainClient.MMRLeafPartial(0, 0, bytes32(0), 0, 0, bytes32(0)),
-        new bytes32[](0),
-        0
-    );
+
+    function getHeaderProof() internal returns (IParachainClient.Proof memory) {
```
Comment (vgeddes) [inline core/packages/contracts/test/InboundQueue.t.sol:25]:
```suggestion
    function makeHeaderProof() internal returns (IParachainClient.Proof memory) {
```

File: core/packages/contracts/.gas-snapshot [hunk 2]
```diff
@@ -1,29 +1,45 @@
...
+SolmateMerkleProofLibTest:testVerifyShortProofSupplied() (gas: 1929)
+SolmateMerkleProofLibTest:testVerifyValidProofSupplied() (gas: 2227)
+TokenVaultTest:testDepositSuccessful() (gas: 108539)
+TokenVaultTest:testInsufficientBalance() (gas: 17294)
+TokenVaultTest:testNonOwnerCannotWithdraw() (gas: 48935)
+TokenVaultTest:testTokenTransferFailedInsufficientAllowance() (gas: 68190)
+TokenVaultTest:testWithdrawSuccessful() (gas: 124731)
+UpgradeProxyTest:testUpgrade() (gas: 30212)
+UpgradeProxyTest:testUpgradeFail() (gas: 20817)
+UpgradeProxyTest:testUpgradeFailBadOrigin() (gas: 10959)
```
Comment (vgeddes) [inline core/packages/contracts/.gas-snapshot:45]: @doubledup after merging these changes, can you submit a PR to foundry to add a final newline for the output of this command?

**Review by yrong — APPROVED** [review]
LGTM!

**Comment by vgeddes** [comment]
Let's hold off on merging this, there will be a merge conflict with some other work I have in the pipeline

**Review by alistair-singh — APPROVED** [review]

**Comment by doubledup** [comment]
This depends on the Solmate proof change (the inbound queue accepting a proof struct in calldata instead of bytes), but the Solmate change has been reverted because there were missing changes in the relayer. I'll create a separate PR with the Solmate proofs, relayer changes and these changes, and add links in the description of each PR (original Solmate PR, this one & the new one).

File: core/packages/contracts/.gas-snapshot (hunk 2 above)
Comment (doubledup) [reply core/packages/contracts/.gas-snapshot:45]: Noted this for after the new Solmate PR 👍

**Pull request closed without merging**